prez-lib/src/types.ts

```diff
@@ -1,183 +1,87 @@
-export interface PrezTerm {
-    dataType?: PrezNode;
-    termType: 'Literal' | 'Node' | 'BlankNode';
-}
+/**
+ * Base type for `PrezLiteral`, `PrezNode` & `PrezBlankNode`
+ */
+export type PrezTerm = {
+    termType: "literal" | "node" | "blanknode";
+};
 
-export interface PrezLiteral extends PrezTerm {
-    termType: 'Literal';
-    text: string;
+/**
+ * Represents an RDF Literal
+ */
+export type PrezLiteral = PrezTerm & {
+    value: string;
+    datatype?: PrezNode;
     language?: string;
-}
+    termType: "literal";
+};
 
-export interface PrezNode extends PrezTerm {
-    termType: 'Node';
+/**
+ * Represents an RDF Named Node
+ */
+export type PrezNode = PrezTerm & {
+    iri: string;
     label?: PrezLiteral;
     description?: PrezLiteral;
-    uri: string;
+    provenance?: PrezLiteral;
     curie?: string;
-}
-
-export interface PrezList {
-    class: PrezTerm;
-    title?: PrezLiteral;
-    description?: PrezLiteral;
-    rows: PrezListRow[];
-    headers: Record<string, PrezNode>;
-}
-
-export interface PrezListRow {
-    [key: string]: PrezTerm;
-}
-
-export interface PrezObject {
-    class: PrezTerm;
-    title: PrezLiteral;
-    description?: PrezLiteral;
-    properties: PrezProperties;
-}
-
-// index each property based on their name/curie for simple access
-export type PrezProperties = Record<string, PrezProperty>;
-
-export interface PrezProperty {
-    predicate: PrezNode;
-    object: PrezTerm|PrezTerm[];
-}
-
-export interface PrezBlankNode extends PrezTerm {
-    termType: 'BlankNode';
-    properties: PrezProperties;
-}
-
```

Comment:
I think there is value in having this data factory method. It gives us a shorthand consistent way to create Prez data objects.

Reply:
`prez-lib` would handle generating these objects in the end, so verbosity isn't really a concern other than for developer convenience. Also, I would think that the conciseness of a data factory would actually make it less clear to read what's in each object for story purposes - i.e. can't see the full object structure. Again, I can add this kind of logic in later if we need it.

Reply:
I think it'll make story construction simpler, and not sure it's unclear? Anyway, can review when getting into the stories later.